### ISSUE TITLE
[ESQL] Version is a required field

### ISSUE THREAD

**Issue by cauemarcondes** [elastic/elasticsearch-js#2247]
On this [PR](https://github.com/elastic/elasticsearch/pull/107433) the ESQL _query api was changed making the `version` property required. 

Currently,  if I call the esql query I got back this error message:
```
{
  name: 'ResponseError',
  message: 'action_request_validation_exception\n' +
    '\tRoot causes:\n' +
    '\t\taction_request_validation_exception: Validation Failed: 1: [version] is required, latest available version is [2024.04.01];'
}
```

How does the query look like now?
```
POST _query
{
  "query": """
  FROM my-index*
  """,
  "version": "2024.04.01"
}
```

We should change the `src/api/api/esql.ts` file to accept a version in the API.

**Comment by stratoula** [comment]
There are 2 ways to run the `_query` in kibana atm:

1. Use the async search strategy https://github.com/elastic/kibana/blob/main/src/plugins/data/server/search/strategies/esql_async_search/esql_async_search_strategy.ts. One example of the usage [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/classes/sources/esql_source/esql_source.tsx)
2. The other is using the es client like [that](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/query/validate_esql_query.ts#L44)

**Comment by JoshMock** [comment]
A default version is now injected if one is not provided. This fix has been merged to main and 8.14 branches, so it will be available when 8.14.0 ships.
